controllers/argocd/deployment_test.go

```diff
@@ -40,6 +40,44 @@ var (
 		"argocd-server"}
 )
 
```

Comment:
Please add a test to verfiy that if ha is enabled, argocd server replica count in cr is ignored.

Reply:
@wtam I've added to check if autoscale is enabled, could you check if what I have now is sufficient?